.github/workflows/test.yml

```diff
@@ -249,6 +249,108 @@ jobs:
         name: kind-logs-${{ env.JOB_NAME }}-${{ github.run_id }}
         path: /tmp/kind/logs
 
+  e2e-dual-conversion:
+    name: e2e-dual-conversion
+    if: github.event_name != 'schedule'
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    strategy:
+      fail-fast: false
+      matrix:
+        gateway-mode: [local, shared]
+    needs: [build]
+    env:
+      JOB_NAME: "DualStack-conversion-${{ matrix.gateway-mode }}"
+      OVN_HA: "true"
+      KIND_IPV4_SUPPORT: "true"
+      KIND_IPV6_SUPPORT: "false"
+      OVN_HYBRID_OVERLAY_ENABLE: "false"
+      OVN_GATEWAY_MODE: "${{ matrix.gateway-mode }}"
+      OVN_MULTICAST_ENABLE:  "false"
+    steps:
+
+    - name: Set up Go
+      uses: actions/setup-go@v1
+      with:
+        go-version: ${{ env.GO_VERSION }}
+      id: go
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+
+    - name: Set up environment
+      run: |
+        export GOPATH=$(go env GOPATH)
+        echo "GOPATH=$GOPATH" >> $GITHUB_ENV
+        echo "$GOPATH/bin" >> $GITHUB_PATH
+
+    - name: Disable ufw
+      # For IPv6 and Dualstack, ufw (Uncomplicated Firewall) should be disabled.
+      # Not needed for KIND deployments, so just disable all the time.
+      run: |
+        sudo ufw disable
+
+    - uses: actions/download-artifact@v2
+      with:
+        name: test-image
+
+    - name: Load docker image
+      run: |
+        docker load --input image.tar
+
+    - name: kind IPv4 setup
+      run: |
+        export OVN_IMAGE="ovn-daemonset-f:dev"
+        make -C test install-kind
+
+    - name: Run Single-Stack Tests
+      run: |
+        make -C test shard-test WHAT="Networking Granular Checks"
+
+    - name: Convert IPv4 cluster to Dual Stack
+      run: |
+        ./contrib/kind-dual-stack-conversion.sh
```

Comment:
The previous commit said "NOT FOR TESTING" in big letters...

Reply:
Tim asked for having a presubmit job and we don't have a better way to do the the dual-stack conversion.
I really don't like to depend on bash scripts for testing, but since it seems the only feasible way :shrug: ... what we should not do is add more things on top this script

Reply:
removed the comment